ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -567,7 +581,13 @@ public boolean hasFailed() {
       }
       // if there is no possible way to use the remaining replicas to meet the success target,
       // deem the operation a failure.
-      return replicaInPoolOrFlightCount + replicaSuccessCount < replicaSuccessTarget;
+      // For Delete, even there is not enough quorum, if offline repair is enabled, continue to run it.
+      if (possibleRunOfflineRepair(replicaInPoolOrFlightCount + replicaSuccessCount)) {
```

Comment:
I suppose we don't have to run this check again, if we know in the constructor that we are going to allow simple operation tracker to continue even if there is no sufficient replicas, we can use a field various to save this flag and use it here rightaway.